### PR TITLE
Add python multi version support

### PR DESCRIPTION
**Pull request by ehooo**
Update CI for python multi-version

Do #70

### CODE REVIEW

**Comment by ehooo** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ehooo** [comment]
recheck

**Comment by ehooo** [comment]
recheck

**Comment by ehooo** [comment]
I have read the Code of Conduct and I hereby accept the Terms

**Comment by ehooo** [comment]
recheck

File: .github/workflows/build-test-release.yml [hunk 1]
```diff
@@ -33,19 +33,29 @@ jobs:
...
         with:
-          python-version: "3.7"
+          python-version: "${{ matrix.python-version }}"
       - uses: pre-commit/action@v3.0.0
   test-unit:
     runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [ 3.7, 3.8, 3.9, 3.10, 3.11 ]
```
Comment (artemrys) [inline .github/workflows/build-test-release.yml:52]: You need to quote "3.10" and "3.11", otherwise GitHub does not recognise it as "3.10", but as "3.1".

**Review by artemrys — CHANGES_REQUESTED** [review]
I think you just need to update "test-unit" job to achieve your gold.

`pre-commit` should give us everything we need with just 1 version of the Python, the same with `publish` job.

File: .github/workflows/build-test-release.yml (hunk 1 above)
Comment (ehooo) [reply .github/workflows/build-test-release.yml:52]: Thanks, I didn't know that issue

**Comment by artemrys** [comment]
@ehooo I made a small change myself, rebased from main branch as well and will approve and merge your PR! Thank you for contribution.

**Pull request merged**